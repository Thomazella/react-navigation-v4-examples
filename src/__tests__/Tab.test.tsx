/**
 * @format
 */

import 'react-native';
import React from 'react';
import { View, Text } from 'react-native';
import { createBottomTabNavigator } from 'react-navigation-tabs';

import { renderWithNavigation } from './testUtils';

const AnotherScreen = () => (
  <View>
    <Text>Another Screen</Text>
  </View>
);

const TestScreen = () => (
  <View>
    <Text>Test Screen</Text>
  </View>
);

describe('TabNavigator', () => {
  it('should render properly', () => {
    const TabNavigator = createBottomTabNavigator({
      Tab1: AnotherScreen,
      Tab2: TestScreen,
    });
    const rendered = renderWithNavigation(TabNavigator);
    expect(rendered.asJSON()).toMatchSnapshot();
  });
});
