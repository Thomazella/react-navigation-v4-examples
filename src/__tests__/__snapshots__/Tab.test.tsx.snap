// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabNavigator should render properly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={false}
        pointerEvents="auto"
        removeClippedSubviews={false}
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "opacity": 1,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View>
            <Text>
              Another Screen
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      pointerEvents="auto"
      style={
        Object {
          "bottom": 0,
          "elevation": 8,
          "left": 0,
          "margin": undefined,
          "marginBottom": undefined,
          "marginHorizontal": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "marginVertical": undefined,
          "position": undefined,
          "right": 0,
          "top": undefined,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "backgroundColor": "#fff",
            "borderTopColor": "#a7a7aa",
            "borderTopWidth": 0.5,
            "flexDirection": "row",
            "height": 49,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          }
        }
      >
        <TouchableWithoutFeedback
          accessibilityLabel="Tab1, tab, 1 of 2"
          accessibilityRole="button"
          accessibilityStates={
            Array [
              "selected",
            ]
          }
          hitSlop={
            Object {
              "bottom": 5,
              "left": 15,
              "right": 15,
              "top": 0,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "flexDirection": "column",
                  "justifyContent": "flex-end",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "height": 29,
                  },
                  false,
                  Object {
                    "flex": 1,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "height": "100%",
                      "justifyContent": "center",
                      "minWidth": 25,
                      "position": "absolute",
                      "width": "100%",
                    },
                    Object {
                      "opacity": 1,
                    },
                  ]
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "height": "100%",
                      "justifyContent": "center",
                      "minWidth": 25,
                      "position": "absolute",
                      "width": "100%",
                    },
                    Object {
                      "opacity": 0,
                    },
                  ]
                }
              />
            </View>
            <Text
              allowFontScaling={true}
              numberOfLines={1}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#007AFF",
                  "fontSize": 11,
                  "marginBottom": 1.5,
                  "textAlign": "center",
                }
              }
            >
              Tab1
            </Text>
          </View>
        </TouchableWithoutFeedback>
        <TouchableWithoutFeedback
          accessibilityLabel="Tab2, tab, 2 of 2"
          accessibilityRole="button"
          accessibilityStates={Array []}
          hitSlop={
            Object {
              "bottom": 5,
              "left": 15,
              "right": 15,
              "top": 0,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "flexDirection": "column",
                  "justifyContent": "flex-end",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "height": 29,
                  },
                  false,
                  Object {
                    "flex": 1,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "height": "100%",
                      "justifyContent": "center",
                      "minWidth": 25,
                      "position": "absolute",
                      "width": "100%",
                    },
                    Object {
                      "opacity": 0,
                    },
                  ]
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "height": "100%",
                      "justifyContent": "center",
                      "minWidth": 25,
                      "position": "absolute",
                      "width": "100%",
                    },
                    Object {
                      "opacity": 1,
                    },
                  ]
                }
              />
            </View>
            <Text
              allowFontScaling={true}
              numberOfLines={1}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#8e8e93",
                  "fontSize": 11,
                  "marginBottom": 1.5,
                  "textAlign": "center",
                }
              }
            >
              Tab2
            </Text>
          </View>
        </TouchableWithoutFeedback>
      </View>
    </View>
  </View>
</View>
`;
